core/src/main/java/org/mule/runtime/core/privileged/transaction/AbstractSingleResourceTransaction.java

```diff
@@ -41,7 +42,7 @@
    *
    * @see javax.transaction.Status
    */
-  protected static Map<Integer, String> txStatusMappings = new HashMap<>(10); // populated later
+  protected static Map<Integer, String> txStatusMappings = new SmallMap<>(); // populated later
```

Comment:
this was already expectin 10 elements, won't this change hurt instead of help?

Reply:
key is an integer. rolling back